### ISSUE TITLE
[P1] Learned-codebook distance cancellation collapses a translated finite point cloud into duplicate codes

### ISSUE THREAD

**Issue by RobTand** [RobTand/tessera#227]
Audited source: ea72d23cbf26a453d0d740afac537fb6d6b8afbc.

_two_means and learn_tree_codebook use default torch.cdist on uncentred float32 values (:68 and :149-151). For these matrix sizes the default norm/GEMM distance expression loses small squared differences when subtracting large nearly equal terms. Assignment can then tie at zero for genuinely distinct centroids. This is a distance evaluation error, not a limitation of the tree nesting model.

Scoped reproduction on torch 2.10.0+cpu:
```python
import torch
from tessera.codebook import _two_means, learn_tree_codebook
for shift in (0.0, 10000.0):
    p = torch.linspace(-1,1,64).view(-1,1) + shift
    print((_two_means(p)-shift).flatten().tolist())
    g = learn_tree_codebook(p, depth=2)
    print(g.values)
```
Unshifted root centroids are approximately [-0.50793648,0.50793648], and the four leaf levels are approximately [0.25396824,0.76190478,-0.76190472,-0.25396821].
At shift 10000 the root centroids relative to the shift become [0,0.5078125]. The four emitted levels are (10000.0,10000.5078125,10000.5078125,10000.5078125): three duplicates and no representation of the lower half, from 64 distinct finite representable inputs.

Expected: translating this same finite cloud should translate its Euclidean fit up to the input's small float32 rounding, not erase a half and create three duplicate leaves. The permitted duplicate-tail fallback does not excuse this: those empty cells were produced by incorrect computed distances.

Scope/impact: learn_tree_codebook is an opt-in free-grid research encoder and these grids are not presently serializable. This is a wrong fitted reconstruction/quality screen on accepted input, not a claim that current serving artifacts are affected. Typical normalized zero-centred inputs may avoid this trigger.

Existing codebook coverage checks _hoist tie semantics in test_audit_container_accounting.py:239, not translated fitting inputs or distance stability.

Fix acceptance: evaluate direct squared differences for each point's own two candidate children, or otherwise centre/choose a numerically stable distance computation. The parent-specific direct pair also avoids materializing the all-node distance matrix. Add the shifted-cloud regression and verify ordinary fitting remains deterministic.

Audit-only finding: not fixed because the user explicitly excluded implementation.

Disposition: filed for a separate implementation agent. The user explicitly requested an audit and issues, with no fixes; no implementation changes were made for this finding.
